### ISSUE TITLE
Nonsticky Rearview

### ISSUE THREAD

**Issue by Mako88** [dxx-rebirth/dxx-rebirth#98]
This has been discussed. Enable an option to allow holding down the rearview button instead of pressing it once, and then again to disable it.

**Comment by vLKp** [comment]
Holding down the rearview button **already** creates a temporary rearview.  I believe the Retro feature is a checkbox that causes all uses of the rearview button to be in this mode, rather than the classic Descent behaviour that short presses lock to rear view and long presses use a temporary rearview.

**Comment by vLKp** [comment]
If this is implemented, I would prefer to do it by splitting the `Rear view` key into three keys: `Rear view`, `Rear stare`, and `Rear glance`.  `Rear view` will work as it always has.  `Rear stare` will always lock to rear view, no matter how long the player holds it.  `Rear glance` will never lock to rear view, no matter how quickly the player releases it.  Users who like the existing time-based overload logic will continue to use `Rear view`.  Users who want the Retro behaviour can change their rear view binding to `Rear glance` or can add a new binding, which would let them use glance or persistent depending on which key they press.  The `Rear stare` option will be for completeness, but also for pilots who want to bind separate keys for glance vs. stare and select the result based on the choice of key pressed instead of the duration of the key press.

Adding such a feature requires modifying the kconfig UI table, which is a bit invasive.

**Comment by zicodxx** [comment]
If people do desire different Rear-view options I think it may be easy enough to provide them. But I am not sure about if the three different options have or should be assigned to three different buttons. If players would like to use different rear-view options at the same time, it makes sense but I imagine just one of the options actually applies to the individual player style (or game mode, Singleplayer, competitive multiplayer).
My idea would be to add a radio-box option in the HUD OPTIONS menu (which doesn't exist yet but will replace the RETICLE menu):
REARVIEW BEHAVIOUR
(  ) ORIGINAL (tap for sticky rearview, hold for temporary)
(  ) STICKY (pressing the key activates rearview, pressing again disables it)
(  ) TEMPORAL (rearview only as long as the key is held down)
This I think adds a great deal of flexibility without messing with the kconfig tables and if people are fine with it, it can stay this way - if not we can still add separate keys.

What do you think?

**Comment by vLKp** [comment]
I considered that, but I am concerned that since D1 players cannot get an inset rear view, they may need sticky rear view in some cases, but want temporal for others.  I use a single pilot file, and thus a single set of preferences, for all my games, regardless of mode.  I suspect many other users also stick to a single file.  Given that, I would stick to original behaviour so that I can get either mode without going into the options menu each time.

I have no objection to adding the radio and gathering feedback from the next release about how best to handle it.

**Comment by zicodxx** [comment]
Yeah, I agree with an option you cannot have all the different behaviours at the same time. But another thing I just remembered: I think I initially broke the original behaviour by removing a specific D1X feature. This allowed to simply specify a "sticky time". With this you could specify how long the rear view button has to be pressed until it becomes temporal. This can enable players to have all behaviours on one button and customize it to their satisfaction.
